qadence/ml_tools/constructors.py

```diff
@@ -733,42 +746,68 @@ def build_qnn_from_configs(
         fm_config (FeatureMapConfig): Feature map configuration.
         ansatz_config (AnsatzConfig): Ansatz configuration.
         observable_config (ObservableConfig): Observable configuration.
+        backend (BackendName): The chosen quantum backend.
+        diff_mode (DiffMode): The differentiation engine to use. Choices are
+            'gpsr' or 'ad'.
+        measurement (Measurements): Optional measurement protocol. If None,
+            use exact expectation value with a statevector simulator.
+        noise (Noise): A noise model to use.
+        configuration (BackendConfiguration | dict): Optional backend configuration.
+        input_diff_mode (InputDiffMode): The differentiation mode for the input tensor.
 
     Returns:
         QNN: A QNN model.
     """
-    fm_blocks = create_fm_blocks(register=register, config=fm_config)
-    full_fm = _interleave_ansatz_in_fm(
-        register=register,
-        fm_blocks=fm_blocks,
-        ansatz_config=ansatz_config,
-    )
 
-    ansatz = create_ansatz(register=register, config=ansatz_config)
+    if observable_config is None:
+        raise ValueError(
+            "You need to provide at least one observable config in the QNN constructor."
+        )
+
+    if ansatz_config is None:
+        ansatz_config = AnsatzConfig(depth=0)
+
+    inputs: list[Basic | str] | None = None
+    full_fm = chain()
```

Comment:
Do you need to instantiate `full_fm` as an empty chain ?

Reply:
Thanks. Removed this to only include `full_fm` in a `blocks` list.